core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -62,8 +62,19 @@ struct FunctorWrapperRangePolicyParallelFor {
       m_functor_wrapper.get_functor()(WorkTag(), id);
   }
 
+  void operator()(sycl::nd_item<1> item) const {
+    const typename Policy::index_type id = item.get_global_id() + m_begin;
+    if (id < m_range + m_begin) {
+      if constexpr (std::is_same<WorkTag, void>::value)
+        m_functor_wrapper.get_functor()(id);
+      else
+        m_functor_wrapper.get_functor()(WorkTag(), id);
+    }
+  }
+
   typename Policy::index_type m_begin;
   FunctorWrapper m_functor_wrapper;
+  typename Policy::index_type m_range = 0;
```

Comment:
Nitpick: I personally would prefer something like `m_work_size` because range for me implies begin and end. It's also a bit weird that it's not used in the `sycl::item<1>` overload as one would expect from a variable named `range`, but I get why as it is only used since the functor may be executed past the end of the actual memory range.

Reply:
[](https://github.com/kokkos/kokkos/pull/4875/commits/df1a78be3e7bb00be6fe7d14533f54a4952974a0)[df1a78b](https://github.com/kokkos/kokkos/pull/4875/commits/df1a78be3e7bb00be6fe7d14533f54a4952974a0)